Classes/Controller/AbstractController.php

```diff
@@ -163,7 +163,14 @@ protected function getClientOutput( \Aimeos\Client\Html\Iface $client ) : string
 
 		$this->response->addAdditionalHeaderData( (string) $client->getHeader() );
```

Comment:
You need the page type for the header too, otherwise you will get mixed header/body combinations

Reply:
True.
The header links (like the canonical for example) then would get mixed up.